var/spack/repos/builtin/packages/chai/package.py

```diff
@@ -39,22 +42,34 @@ class Chai(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.8:', type='build')
     depends_on('cmake@3.9:', type='build', when="+cuda")
 
-    depends_on('blt@0.4.0:', type='build', when='@2.3.1:')
+    depends_on('blt@0.4.1', type='build', when='@2.4.0:')
     depends_on('blt@:0.3.6', type='build', when='@:2.3.0')
 
     depends_on('umpire')
-    depends_on('raja', when="+raja")
-
+    depends_on('umpire@6.0.0', when="@2.4.0")
+    depends_on('umpire@main', when='@main')
     depends_on('umpire+cuda', when="+cuda")
-    depends_on('raja+cuda', when="+raja+cuda")
-
-    # variants +rocm and amdgpu_targets are not automatically passed to
-    # dependencies, so do it manually.
+    for sm_ in CudaPackage.cuda_arch_values:
+        depends_on('umpire+cuda cuda_arch={0}'.format(sm_),
+                   when='cuda_arch={0}'.format(sm_))
     depends_on('umpire+rocm', when='+rocm')
-    depends_on('raja+rocm', when="+raja+rocm")
     for val in ROCmPackage.amdgpu_targets:
         depends_on('umpire amdgpu_target=%s' % val, when='amdgpu_target=%s' % val)
-        depends_on('raja amdgpu_target=%s' % val, when='+raja amdgpu_target=%s' % val)
+
+    with when('+raja'):
+        depends_on('raja')
+        depends_on('raja@0.14.0', when="@2.4.0")
+        depends_on('raja@main', when='@main')
+        with when('+cuda'):
+            depends_on('raja+cuda')
+            for sm_ in CudaPackage.cuda_arch_values:
+                depends_on('raja+cuda cuda_arch={0}'.format(sm_),
+                        when='cuda_arch={0}'.format(sm_))
```

Comment:
looks like these `when` args need to scoot over?

Reply:
hey man, don't criticize my style 😆

Reply:
I am just eagerly awaiting these new versions :-)

Reply:
Totally - I'm just doing some testing running through develop, main, latest numeric to make sure everything concretizes and builds as expected. Should be done by tomorrow